src/scenes/City.ts

```diff
@@ -0,0 +1,28 @@
+import { ICity } from "./ICity";
+export class City implements ICity {
+    // economy= [stock, consume, production]
+    constructor(private name: string, private economy: number[]) {
+        this.name = name;
+        this.economy = economy;
+    }
+
+    private getStock() {
+        return this.economy[0];
+    }
+
+    private getConsume() {
+        return this.economy[1];
+    }
+
+    private getFactories() {
+        return this.getFactories[2];
```

Comment:
I have no idea why typesript doesn't report this as wrong. This calls the third element  of the _method_ `getFactories()`.

Reply:
You probably wanted to write `this.economy[2]` or even better: `this.production`.